sources/packages/backend/apps/api/src/route-controllers/report/report.controller.service.ts

```diff
@@ -1,20 +1,88 @@
-import { Injectable } from "@nestjs/common";
-import { getFileNameAsCurrentTimestamp } from "@sims/utilities";
+import {
+  BadRequestException,
+  Injectable,
+  UnprocessableEntityException,
+} from "@nestjs/common";
+import {
+  ReportService,
+  FILTER_PARAMS_MISMATCH,
+  REPORT_CONFIG_NOT_FOUND,
+} from "@sims/services";
+import {
+  getFileNameAsCurrentTimestamp,
+  CustomNamedError,
+} from "@sims/utilities";
 import { Response } from "express";
 import { Readable } from "stream";
+import { FormService } from "../../services";
+import { ReportsFilterAPIInDTO } from "./models/report.dto";
+import { FormNames } from "../../services/form/constants";
 
 /**
- * Service layer for reports.
+ * Controller Service layer for reports.
  */
 @Injectable()
 export class ReportControllerService {
+  constructor(
+    private readonly reportService: ReportService,
+    private readonly formService: FormService,
+  ) {}
+
+  /**
+   * Generates report in csv format.
+   * @param payload report filter payload.
+   * @param response http response as file.
+   * @param options related options.
+   * - `institutionId` related institution id.
+   */
+  async generateReport(
+    payload: ReportsFilterAPIInDTO,
+    response: Response,
+    options?: { institutionId?: number },
+  ): Promise<void> {
+    const submissionResult = await this.formService.dryRunSubmission(
+      FormNames.ExportFinancialReports,
+      payload,
+    );
+    if (!submissionResult.valid) {
+      throw new BadRequestException(
+        "Not able to export report due to an invalid request.",
+      );
+    }
+    options?.institutionId
+      ? (submissionResult.data.data.params = {
+          ...submissionResult.data.data.params,
+          institutionId: options?.institutionId,
+        })
+      : null;
```

Comment:
Can it be changed as below?
```ts
if (options?.institutionId) {
  submissionResult.data.data.params.institutionId = options.institutionId;
}
```
If the `null` is required it could be as below.
```ts
submissionResult.data.data.params.institutionId = options?.institutionId ?? null;
```